x-pack/legacy/plugins/apm/public/components/app/ErrorGroupOverview/index.tsx

```diff
@@ -22,73 +23,124 @@ import { ErrorDistribution } from '../ErrorGroupDetails/Distribution';
 import { ErrorGroupList } from './List';
 import { useUrlParams } from '../../../hooks/useUrlParams';
 import { useTrackPageview } from '../../../../../infra/public';
+import { PROJECTION } from '../../../projections/typings';
+import { LocalUIFilters } from '../../shared/LocalUIFilters';
+import { useTransactionTypes } from '../../../hooks/useTransactionTypes';
 
 const ErrorGroupOverview: React.SFC = () => {
+  const { urlParams, uiFilters } = useUrlParams();
+
   const {
-    urlParams: { serviceName, start, end, sortField, sortDirection },
-    uiFilters
-  } = useUrlParams();
+    serviceName,
+    start,
+    end,
+    sortField,
+    sortDirection,
+    transactionType
+  } = urlParams;
 
   const { data: errorDistributionData } = useFetcher(() => {
     if (serviceName && start && end) {
       return loadErrorDistribution({
         serviceName,
+        transactionType,
         start,
         end,
         uiFilters
       });
     }
-  }, [serviceName, start, end, uiFilters]);
+  }, [serviceName, start, end, uiFilters, transactionType]);
 
   const { data: errorGroupListData } = useFetcher(() => {
     if (serviceName && start && end) {
       return loadErrorGroupList({
         serviceName,
+        transactionType,
         start,
         end,
         sortField,
         sortDirection,
         uiFilters
       });
     }
-  }, [serviceName, start, end, sortField, sortDirection, uiFilters]);
+  }, [
+    serviceName,
+    start,
+    end,
+    sortField,
+    sortDirection,
+    uiFilters,
+    transactionType
+  ]);
+
+  const transactionTypes = useTransactionTypes(urlParams);
 
-  useTrackPageview({ app: 'apm', path: 'error_group_overview' });
+  useTrackPageview({
+    app: 'apm',
+    path: 'error_group_overview'
+  });
   useTrackPageview({ app: 'apm', path: 'error_group_overview', delay: 15000 });
 
+  const localUIFiltersConfig = useMemo(
+    () => ({
+      filterNames: [
+        'transactionResult',
+        'host',
+        'containerId',
+        'podId'
+      ] as LocalUIFilterName[],
+      params: {
+        serviceName,
+        transactionType
+      },
+      projection: PROJECTION.ERROR_GROUPS
+    }),
+    [serviceName, transactionType]
+  );
```

Comment:
I commented previously about moving this to component props. I still think that's the best approach and then only memoize what needs to be memoized (eg. no need to memoize `projection` like now).
If need be `filterNames` can be declared as a top-level const outside the component and won't need memoization at all. And the question is whether we really need that in the first place. Is `LocalUIFilters` very expensive to render?

Reply:
@sqren it's not about CPU, it's about data fetching. see https://github.com/elastic/kibana/pull/41588#discussion_r310154299.

Would be happy to move out some of the props which do not need to be memoized, but one could also argue that it's nicer to have the complete configuration in one place, instead of one prop outside of the component, one inside of React.memo, and one passed as a prop directly. Thoughts?

Reply:
Ah, missed your answer there. I see, that's annoying 🤔
I can see the advantage of keeping it co-located like you do in that case.